src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -345,13 +365,16 @@ def translate_symbol_ref(
     else:
         sym_value = str(node.id)
         if sym_value in let_symbols:
+            let_node, data_type = let_symbols[sym_value]
+            if isinstance(let_node, gtir.Literal):
+                return sdfg_builder.visit(let_node)
```

Comment:
I have added back a return, yes, but I am returning the output of the sdfg visitor (remember, initially let_symbols contained access nodes and I was immediately returning the access node, because I assumed it belonged to current state). The new return is a bypass, it will call the same function but enter a different path below. Maybe it is confusing that I am using the same primitive translator for both `SymRef` and `Literal`.

Reply:
Okay this makes sense.
It might be strange a little bit, I would put a note to say that it is intentional, such that nobody thinks it might be a bug.
Btw. the doc string only mention `SymRef` and not `Literals`.